### DISCUSSION TITLE
Persist Plot Panels when Switching Tabs and Layouts

### DISCUSSION THREAD

**Discussion by oosthoek** [foxglove/community#846] [Q&A]
Hello the plot data reset whenever switching tabs or layouts.  Can there be an option to persist it?

**Comment by jtbandes** [comment]
See also: https://github.com/orgs/foxglove/discussions/842

This option would have a lot of implications for bandwidth and memory usage, but does seem like it could be helpful for some users.

**Comment by Pandananana** [comment]
This feature is a must have for me! Need it so I can see logs if something goes wrong, but I don't want the logs always taking up space on the main tab

**Comment by karl-schulz** [comment]
Is there anything we can do to have this behavior?

**Reply by defunctzombie** [reply]
There's nothing to work around this for the live playback workflow. For local files or cloud files you see the entire history.

**Comment by JMarple** [comment]
This is definitely a much desired feature on our team. 

If memory & performance are the key concern when implementing this, maybe there could be a setting for a maximum buffer size?

**Reply by defunctzombie** [reply]
Specifically for live connectivity workflows?

**Reply by JMarple** [reply]
Correct, when using live connections over the Foxglove websocket.  

When we're reading from a stored file on the Foxglove data platform, or a local file, data persists between tabs as expected (though it can take a second or two to re-load).

**Comment by Masterzach32** [comment]
This is also a very desired feature for our team. It doesn't need to be enabled by default or for all plots, but for some plots it would be very useful to keep the live history and continue to update when switching to other tabs.
